pkg/manager/impl/execution_manager.go

```diff
@@ -428,41 +428,71 @@ func (m *ExecutionManager) getInheritedExecMetadata(ctx context.Context, request
 	return parentNodeExecutionID, sourceExecutionID, nil
 }
 
+// WorkflowExecutionConfigInterface is used as common interface for capturing the common behavior catering to the needs
+// of fetching the WorkflowExecutionConfig across LaunchPlanSpec, ExecutionCreateRequest
+// MatchableResource_WORKFLOW_EXECUTION_CONFIG and ApplicationConfig
+type WorkflowExecutionConfigInterface interface {
```

Comment:
why does this need to be an interface?

Reply:
It becomes very convenient now to use the same merge function 

```mergeIntoExecConfig(workflowExecConfig *admin.WorkflowExecutionConfig, spec WorkflowExecutionConfigInterface)```

for launchplan spec, executionCreateRequest, matchable resorce and application config.
Without these we will have to define individual functions for each of these object types.

Abstracting this common functionality across all these make it easier to manage the code.
Let me know if you think otherwise

Reply:
I see, maybe one day https://github.com/flyteorg/flyte/issues/2242 will simplify this!

Reply:
Yeah that would help simplify a lot. I couldn't find a way to do this with current protobuf and faced the same issue with another service we wrote where we had to duplicate the message fields .